pkg/controller/machine/machine.go

```diff
@@ -307,42 +307,34 @@ func (c *Controller) getNodeByNodeRef(nodeRef *corev1.ObjectReference) (*corev1.
 }
 
 func (c *Controller) updateMachine(machine *clusterv1alpha1.Machine, modify func(*clusterv1alpha1.Machine)) (*clusterv1alpha1.Machine, error) {
-	var updatedMachine *clusterv1alpha1.Machine
-
 	// Both machine and updatedMachine can be nil later on, so we store the namespace and name here
 	namespace := machine.Namespace
 	name := machine.Name
 	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		var getErr, retryErr error
+		var err error
 
-		// Apply modifications
-		// machine could be nil here if the get after a failed update failed
-		if machine == nil {
-			machine, getErr = c.machineClient.ClusterV1alpha1().Machines(namespace).Get(name, metav1.GetOptions{})
-			if getErr != nil {
-				return getErr
-			}
+		machine, err = c.machineClient.ClusterV1alpha1().Machines(namespace).Get(name, metav1.GetOptions{})
```

Comment:
The drawback of this is that we always do at least one GET, but IMHO this is negligible, compared to the simplification. Also this drawback will vanish once we switch to controller-runtime, as the clients reader interface will then be lister-backed.

Reply:
Why don't we use the lister?

Reply:
Oh wow, yeah ofc. That one was too obvious.